### DISCUSSION TITLE
How does this card get to know the position in coordinates the user clicks on the screen?

### DISCUSSION THREAD

**Discussion by obaldius** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#446] [Q&A] [answered]
I'd like to built my own spot clean in my dashboard with my buttons, images and so on. How does this card get to know which position in coordinates de user clicked the screen? How can I adapt that to my own, let's say, picture card or whatever?

thanks

**Comment by PiotrMachowski** [comment, marked as answer]
It uses affine/projective transformations to calculate mapping between screen and vacuum coordinates. I don't think you can achieve it using picture-elements. You can configure room cleaning though (if your vacuum supports it)

**Reply by obaldius** [reply]
Thanks mate, you saved me a lot of time.
